### PR TITLE
Move WP_REST_Block_Navigation_Areas_Controller from Gutenberg to Core.

### PR DESCRIPTION
**Pull request by anton-vlasenko**
## Description
This PR aims to change the route of the `__experimental/block-navigation-areas` REST API endpoint.
It has to be renamed to `/wp/v2/block-navigation-areas` because we have to merge the `WP_REST_Block_Navigation_Areas_Controller` controller into Core.
Also, this PR enables batching for this new (old) REST route.

## How has this been tested?
1. Go to `Gutenberg -> Experiments` and enable `Navigation Screen`.
2. Go to `Gutenberg -> Navigation` page.
3. Open the console in your browser, and execute the following code:
```js
wp.apiFetch( { path: '/wp/v2/block-navigation-areas' } ).then( ( posts ) => {
    console.log( posts );
} );
```
4. You should see a valid response in the console (HTTP status 200).

## Screenshots 
No screenshots are needed.

## Types of changes
New feature

## Checklist:
- [ ] My code is tested.
- [ ] My code follows the WordPress code style. 
- [ ] My code follows the accessibility standards. 
- [ ] I've tested my changes with keyboard and screen readers. 
- [ ] My code has proper inline documentation. 
- [ ] I've included developer documentation if appropriate. 
- [ ] I've updated all React Native files affected by any refactorings/renamings in this PR (please manually search all `*.native.js` files for terms that need renaming or removal).

### CODE REVIEW

**Comment by spacedmonkey** [comment]
@anton-vlasenko Do you want a review for this yet?

**Comment by anton-vlasenko** [comment]
@spacedmonkey Yes, please :)

File: lib/class-wp-rest-block-navigation-areas-controller.php [hunk 1]
```diff
@@ -37,7 +37,8 @@ public function register_routes() {
 					'permission_callback' => array( $this, 'get_items_permissions_check' ),
 					'args'                => $this->get_collection_params(),
 				),
-				'schema' => array( $this, 'get_public_item_schema' ),
+				'schema'      => array( $this, 'get_public_item_schema' ),
+				'allow_batch' => array( 'v1' => true ),
```
Comment (spacedmonkey) [inline lib/class-wp-rest-block-navigation-areas-controller.php:41]: We are now following this pattern. 

https://github.com/WordPress/wordpress-develop/commit/ecf1d6a158c70dd7f10cc42c4b26ecb0fb22bb59

So this should be 
```
'allow_batch' => $this->allow_batch,
```

**Comment by spacedmonkey** [comment]
I posted some comments on https://github.com/WordPress/gutenberg/pull/36178. Like someone to look at those comments.

**Comment by noisysocks** [comment]
Going to merge this now so as to fix CI which is failing now that Core has the endpoint committed.

**Review by noisysocks — APPROVED** [review]

File: lib/class-wp-rest-block-navigation-areas-controller.php (hunk 1 above)
Comment (anton-vlasenko) [reply lib/class-wp-rest-block-navigation-areas-controller.php:41]: Good to know.
This task is already merged, but I will be using `'allow_batch' => $this->allow_batch` from now on.

**Pull request merged**